fearless/Common/Storage/PersistentValueSettings.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+
+class PersistentValueSettings<T> {
+    let storageFacade: StorageFacadeProtocol
+
+    init(storageFacade: StorageFacadeProtocol) {
+        self.storageFacade = storageFacade
+    }
+
+    private let mutex = NSLock()
+
+    var internalValue: T?
+
+    var value: T! {
+        mutex.lock()
+
+        defer {
+            mutex.unlock()
+        }
+
+        return internalValue
+    }
+
+    var hasValue: Bool { value != nil }
+
+    func performSetup(completionClosure _: @escaping (Result<T?, Error>) -> Void) {
+        // Function must be implemented in subclass
```

Comment:
Isn't it safer to throw an error here?

Reply:
Yep, we could add `fatalError("Must be implemented on inheritance")`. Will do in the next PR.